packages/thirdweb/src/extensions/erc4626/read/asset.macro.ts

```diff
@@ -0,0 +1,51 @@
+import { readContract } from "../../../transaction/read-contract.js";
+import type { BaseTransactionOptions } from "../../../transaction/types.js";
+/**
+ * @macro delete-next-lines
+ */
+import { prepareMethod } from "../../../utils/abi/prepare-method.js";
+import { $run$ } from "@hazae41/saumon";
+import type { Address } from "abitype";
+
+/**
+ * Represents the parameters for retrieving the underlying asset of a tokenized vault.
+ */
+type AssetParams = {
```

Comment:
should be exported so it can be used by (eventual) consumers to extend their own inputs